FontLoader.js

```diff
@@ -337,22 +408,35 @@
 			}
 			
 			if (this._numberOfLoadedFonts < this._numberOfFontFamilies) {
-				for (fontFamily in this._fontsMap) {
-					if (this._fontsMap.hasOwnProperty(fontFamily)) {
-						notLoadedFontFamilies.push(fontFamily);
+				for (font in this._fontsMap) {
+					if (this._fontsMap.hasOwnProperty(font)) {
+						notLoadedFontFamilies.push(this._fontsMap[font]);
 					}
 				}
+
 				callbackParameter = {
-					message: "Not all fonts were loaded",
+					message: "Not all fonts were loaded (" + this._numberOfLoadedFonts + "/" + this._numberOfFontFamilies + ")",
 					notLoadedFontFamilies: notLoadedFontFamilies
 				};
 			} else {
 				callbackParameter = null;
 			}
 			if (this.delegate && typeof this.delegate.fontsLoaded === "function") {
-				this.delegate.fontsLoaded(callbackParameter);
+				// let the browser process the fonts, this could prevent some race-conditions
+				setTimeout(function() {
```

Comment:
Why `setTimeout` here? Why should there be race conditions in `_finish` function?
This function is already race-condition-protected.